### PR TITLE
Retry if response is not success

### PR DESCRIPTION
**Pull request by limhm1396**

### CODE REVIEW

File: src/kross-client/base.ts [hunk 1]
```diff
@@ -31,6 +37,10 @@ export class KrossClientBase {
 
     this.instance.interceptors.request.use(
       async (config: AxiosRequestConfig) => {
+        if (config.url?.includes('/auth/refresh')) {
```
Comment (Jscripter-pk) [inline src/kross-client/base.ts:40]: @limhm1396 I don't think these line will add any value instead they will keep failing as till there we don't have HMAC header as well as authorization ones

File: src/kross-client/base.ts [hunk 2]
```diff
@@ -105,6 +115,7 @@ export class KrossClientBase {
               this.refreshToken = null;
               if (this?.forceLogoutCallback) await this.forceLogoutCallback();
             }
+            return config;
```
Comment (Jscripter-pk) [inline src/kross-client/base.ts:118]: @limhm1396 If we return here or not I don't think it also make any thing improve.

**Comment by limhm1396** [comment]
@Jscripter-pk sure :)

**Comment by limhm1396** [comment]
@Jscripter-pk OK, you're right. kross-sdk is also used mobile and orange too. So if change this, unexpected something can happen. I think it could be better to close this issue. I'll update code on cross-meta. If you think same, could you please close this issue?

**Comment by Jscripter-pk** [comment]
@limhm1396 I think it's Ok if you just add retry as you did in response interceptor but try not to make change so that we don't get any unexpected behaviour in authentication and authorisation.

**Comment by limhm1396** [comment]
@Jscripter-pk How do you think about we only retry request when we receive 5xx error?

**Comment by Jscripter-pk** [comment]
@limhm1396 it is now upto you what use-case you are specifically trying to resolve. If it is specifically for 5xx then you are good to go

File: src/kross-client/base.ts (hunk 1 above)
Comment (limhm1396) [reply src/kross-client/base.ts:40]: @Jscripter-pk How can I handle without this condition? refresh token header is different normal request.

**Review by Jscripter-pk — APPROVED** [review]

**Comment by Jscripter-pk** [comment]
@limhm1396 resolve the conflict and merge

**Pull request merged**